### ISSUE TITLE
Trainer does not change TF Variables?

### ISSUE THREAD

**Issue by wielandbrendel** [tflearn/tflearn#266]
This is probably more an issue of me understanding TFLearn. However, I noticed that training a network using tflearn.trainer does not change the values of the variables even though a decrease in the loss is reported. A simple toy example with a single scalar weight being trained is here:

```
import tensorflow as tf
import tflearn
import numpy as np

sess = tf.InteractiveSession()

input_var = tf.placeholder(dtype=tf.float32, shape=(None, 1))
inp      = tflearn.input_data(placeholder=input_var, name='Input_layer')
enc      = tflearn.fully_connected(inp, n_units=1, activation='ReLU', name='Encoding_1')
loss     = tf.reduce_mean((enc - 3*inp)**2)

sess.run(tf.initialize_all_variables())

print(sess.run(tf.trainable_variables()))

optimizer = tf.train.AdamOptimizer(1e-3)
trainop   = tflearn.TrainOp(loss=loss,
                            optimizer=optimizer,
                            metric=None,
                            batch_size=100)

trainer = tflearn.Trainer(train_ops=trainop)

trainer.fit(feed_dicts={input_var : np.random.randn(100000,1).astype(np.float32)},
            show_metric=True,
            n_epoch=2, run_id='beta')

print(sess.run(tf.trainable_variables()))
```
Before and after training the weights are the same despite a decline in the regression loss (as reported during training)! This is a completely counter-intuitive behaviour for me. Could somebody explain to me what is going on? Thanks!

**Comment by aymericdamien** [comment]
The reason is that Trainer object has its own session, so to make it works, you have to use that session (instead of the one define above).
Use: `trainer.session` instead of `sess`.

**Comment by wielandbrendel** [comment]
Or one explicitly passes a session object as an argument to trainer. Thanks for the clarification! Just discovered that you also have a Gitter chat, so I am likely bothering you there with more questions ;-).
